.github/workflows/docker-publish.yml

```diff
@@ -0,0 +1,59 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+name: Publish Docker images
+
+concurrency:
+  group: ${{ github.repository }}-${{ github.head_ref || github.sha }}-${{ github.workflow }}
+  cancel-in-progress: true
+
+on:
+  push:
+    tags:
+      - '*.*.*'
+      - '*.*.*-rc*'
+
+docker:
+  name: Docker
+  runs-on: ubuntu-latest
+  permissions:
+    contents: read
+    packages: write
+  steps:
+    - uses: actions/checkout@v3
+    - name: Set up Java
+        uses: actions/setup-java@v3
+        with:
+          java-version: '17'
+    - name: Extract Maven version
+      id: extract_version
+      run: |
+        COMET_VERSION=$(mvn help:evaluate -Dexpression=project.version -q -DforceStdout)
+        echo "COMET_VERSION=$COMET_VERSION" >> $GITHUB_ENV
+    - name: Echo Maven version
+      run: echo "The current Maven version is ${{ env.COMET_VERSION }}"
+    - name: Build and push Docker image
+      run: |
+        docker build -t datafusion-comet:spark-3.4-scala-2.12-${{ env.COMET_VERSION }} -f kube/Dockerfile .
```

Comment:
are we planning to parametrize spark and scala versions later? the same like its done for comet version

Reply:
good point, I will update that

Reply:
I think also it might be important here to use docker buildx to create images that are compatible with arm if comet supports it

Reply:
Thanks @edmondop I have made this change